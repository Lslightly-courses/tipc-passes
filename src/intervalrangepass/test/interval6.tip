main() {
    var y, x;
    y = 0;
    x = 7;
    x = x+1;
    while (input) { // x = [8,8], y = [0,+inf]
        x = 7;
        x = x+1;
        y = y+1;
    }
    return x;
}