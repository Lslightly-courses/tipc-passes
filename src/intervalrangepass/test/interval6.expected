*** interval range analysis for function define i64 @_tip_main() {
entry:
  %addtmp = add i64 7, 1
  br label %header1

header1:                                          ; preds = %body1, %entry
  %y.0 = phi i64 [ 0, %entry ], [ %addtmp5, %body1 ]
  %x.0 = phi i64 [ %addtmp, %entry ], [ %addtmp3, %body1 ]
  %0 = call i64 @_tip_input()
  %loopcond = icmp ne i64 %0, 0
  br i1 %loopcond, label %body1, label %exit1

body1:                                            ; preds = %header1
  %addtmp3 = add i64 7, 1
  %addtmp5 = add i64 %y.0, 1
  br label %header1

exit1:                                            ; preds = %header1
  call void @llvm.donothing()
  ret i64 %x.0
}
 ***
  %addtmp = add i64 7, 1 = [8,8]
  %0 = call i64 @_tip_input() = [-inf,+inf]
  %addtmp3 = add i64 7, 1 = [8,8]
  %y.0 = phi i64 [ 0, %entry ], [ %addtmp5, %body1 ] = [0,+inf]
  call void @llvm.donothing() = [-inf,+inf]
  %x.0 = phi i64 [ %addtmp, %entry ], [ %addtmp3, %body1 ] = [8,8]
  %addtmp5 = add i64 %y.0, 1 = [1,+inf]
  %loopcond = icmp ne i64 %0, 0 = [0,1]
